From 7ef99ff2dbeb07c3452db740bb41acb792e14418 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Mon, 20 Mar 2017 16:04:15 +0800
Subject: [PATCH 1601/5242] MLK-14498-6 tty: serial: imx: pass ->dev to
 dma_alloc_coherent() API

commit  6d84d0da94b10bdee94cc542317ec30e164abdde from
https://source.codeaurora.org/external/imx/linux-imx.git

Pass ->dev to dma_alloc_coherent() API to avoid kernel dump.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 1bde76b..4bc4e9f 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1241,7 +1241,7 @@ static void imx_uart_dma_exit(struct imx_port *sport)
 		sport->dma_chan_rx = NULL;
 
 		if (sport->rx_buf.buf) {
-			dma_free_coherent(NULL, IMX_RXBD_NUM * RX_BUF_SIZE,
+			dma_free_coherent(sport->port.dev, IMX_RXBD_NUM * RX_BUF_SIZE,
 						(void *)sport->rx_buf.buf,
 						sport->rx_buf.dmaaddr);
 			sport->rx_buf.buf = NULL;
@@ -1278,7 +1278,7 @@ static int imx_uart_dma_init(struct imx_port *sport)
 		goto err;
 	}
 
-	sport->rx_buf.buf = dma_alloc_coherent(NULL, IMX_RXBD_NUM * RX_BUF_SIZE,
+	sport->rx_buf.buf = dma_alloc_coherent(sport->port.dev, IMX_RXBD_NUM * RX_BUF_SIZE,
 					&sport->rx_buf.dmaaddr, GFP_KERNEL);
 	if (!sport->rx_buf.buf) {
 		dev_err(dev, "cannot alloc DMA buffer.\n");
-- 
1.7.9.5

